src/Interpreters/PredicateRewriteVisitor.cpp

```diff
@@ -107,26 +104,14 @@ bool PredicateRewriteVisitorData::rewriteSubquery(ASTSelectQuery & subquery, con
         || hasNonRewritableFunction(subquery.select(), getContext()))
         return false;
 
-    Names outer_columns = table_columns.columns.getNames();
     for (const auto & predicate : predicates)
     {
         std::vector<ASTIdentifier *> identifiers;
         ASTPtr optimize_predicate = predicate->clone();
         cleanAliasAndCollectIdentifiers(optimize_predicate, identifiers);
 
         for (const auto & identifier : identifiers)
-        {
             IdentifierSemantic::setColumnShortName(*identifier, table_columns.table);
-            const auto & column_name = identifier->name();
-
-            /// For lambda functions, we can't always find them in the list of columns
-            /// For example: SELECT * FROM system.one WHERE arrayMap(x -> x, [dummy]) = [0]
-            const auto & outer_column_iterator = std::find(outer_columns.begin(), outer_columns.end(), column_name);
-            if (outer_column_iterator != outer_columns.end())
-            {
-                identifier->setShortName(inner_columns[outer_column_iterator - outer_columns.begin()]);
-            }
-        }
```

Comment:
Could you, please, explain a little bit, why this code is not needed anymore ...
And where the bug was.